docs/csharp/tour-of-csharp/index.md

```diff
@@ -0,0 +1,86 @@
+ms.technology: devlang-csharp
+ms.devlang: csharp
+ms.assetid: ebc727cd-8112-42e7-b59c-3c2873ad661c
+---
+
+# A Tour of the C# Language  
+
+C# (pronounced “See Sharp”) is a simple, modern, object-oriented, and type-safe programming language. C# has its roots in the C family of languages and will be immediately to C, C++, Java, and JavaScript programmers.
+
+C# is an object-oriented language, but C# further includes support for ***component-oriented*** programming. Contemporary software design increasingly relies on software components in the form of self-contained and self-describing packages of functionality. Key to such components is that they present a programming model with properties, methods, and events; they have attributes that provide declarative information about the component; and they incorporate their own documentation. C# provides language constructs to support directly these concepts, making C# a very natural language in which to create and use software components.
+
+Several C# features aid in the construction of robust and durable applications: ***Garbage collection*** automatically reclaims memory occupied by unreachable unused objects; ***exception handling*** provides a structured and extensible approach to error detection and recovery; and the ***type-safe*** design of the language makes it impossible to read from uninitialized variables, to index arrays beyond their bounds, or to perform unchecked type casts.
+
+C# has a ***unified type system***. All C# types, including primitive types such as `int` and `double`, inherit from a single root `object` type. Thus, all types share a set of common operations, and values of any type can be stored, transported, and operated upon in a consistent manner. Furthermore, C# supports both user-defined reference types and value types, allowing dynamic allocation of objects as well as in-line storage of lightweight structures.
+
+To ensure that C# programs and libraries can evolve over time in a compatible manner, much emphasis has been placed on ***versioning*** in C#’s design. Many programming languages pay little attention to this issue, and, as a result, programs written in those languages break more often than necessary when newer versions of dependent libraries are introduced. Aspects of C#’s design that were directly influenced by versioning considerations include the separate `virtual` and `override` modifiers, the rules for method overload resolution, and support for explicit interface member declarations.
+
+## Hello world
+
+The “Hello, World” program is traditionally used to introduce a programming language. Here it is in C#:
+
+```csharp
+using System;
+class Hello
+{
+    static void Main()
+	{
+        Console.WriteLine("Hello, World");
+	}
+}
+```
+
+C# source files typically have the file extension `.cs`. Assuming that the “Hello, World” program is stored in the file `hello.cs`, the program might be compiled using the command line:
+
+```
+csc hello.cs
```

Comment:
newbie question: isn't csc for framework compilation? if so, do we need to call that out?